lib/rules/no-extra-parens.js

```diff
@@ -860,6 +872,22 @@ module.exports = {
                 if (isReturnAssignException(node)) {
                     return;
                 }
+
+                const availableTypes = new Set(["BinaryExpression", "LogicalExpression", "SequenceExpression"]);
+
+                if (isCondTernaryException(node)) {
+                    if (!availableTypes.has(node.test.type) && isParenthesised(node.test)) {
+                        report(node.test);
+                    }
+                    if (!availableTypes.has(node.consequent.type) && isParenthesised(node.consequent)) {
+                        report(node.consequent);
+                    }
+                    if (!availableTypes.has(node.alternate.type) && isParenthesised(node.alternate)) {
+                        report(node.alternate);
+                    }
+                    return;
+                }
```

Comment:
This is still a false positive:

```js
/* eslint no-extra-parens: ["error", "all", { "ternaryOperandBinaryExpressions": false }] */

(a = b) ? c : d; // false positive, removing parentheses will change semantics to a = (b ? c : d)

```

Note that this is just one example where the added `if (isCondTernaryException(node))` branch doesn't work correctly. We should update the code to fall into existing logic when the new option doesn't apply, as suggested in https://github.com/eslint/eslint/pull/17270#discussion_r1230878847.

Reply:
Sorry, I misunderstood this.